packages/pigeon/bin/run_tests.dart

```diff
@@ -0,0 +1,194 @@
+  'dart_compilation_tests': _TestInfo(
+      function: _runDartCompilationTests,
+      description: 'Compilation tests on generated Dart code.'),
+  'dart_unittests': _TestInfo(
+      function: _runDartUnitTests,
+      description: 'Unit tests on and analysis on Pigeon\'s implementation.'),
+  'flutter_unittests': _TestInfo(
+      function: _runFlutterUnitTests,
+      description: 'Unit tests on generated Dart code.'),
+  'ios_e2e_tests': _TestInfo(
+      function: _runIosE2eTests,
+      description: 'End-to-end objc tests run on iOS Simulator'),
+  'ios_unittests': _TestInfo(
+      function: _runIosUnitTests,
+      description: 'Unit tests on generated Objc code.'),
+  'mock_handler_tests': _TestInfo(
+      function: _runMockHandlerTests,
+      description: 'Unit tests on generated Dart mock handler code.'),
+};
+
+Future<Process> _streamOutput(Future<Process> processFuture) async {
+  final Process process = await processFuture;
+  stdout.addStream(process.stdout);
+  stderr.addStream(process.stderr);
+  return process;
+}
+
+Future<int> _runAndroidUnitTests() async {
+  throw UnimplementedError('See run_tests.sh.');
+}
+
+Future<int> _runDartCompilationTests() async {
+  throw UnimplementedError('See run_tests.sh.');
+}
+
+Future<int> _runDartUnitTests() async {
+  throw UnimplementedError('See run_tests.sh.');
+}
+
+Future<int> _runFlutterUnitTests() async {
+  throw UnimplementedError('See run_tests.sh.');
+}
+
+Future<int> _runIosE2eTests() async {
+  throw UnimplementedError('See run_tests.sh.');
+}
+
+Future<int> _runIosUnitTests() async {
+  throw UnimplementedError('See run_tests.sh.');
+}
+
+Future<int> _runMockHandlerTests() async {
+  throw UnimplementedError('See run_tests.sh.');
+}
+
+Future<int> _runPigeon({
+  required String input,
+  String? cppHeaderOut,
+  String? cppSourceOut,
+}) async {
+  const bool hasDart = false;
+  final List<String> args = <String>[
+    'pub',
+    'run',
+    'pigeon',
+    '--input',
+    input,
+  ];
+  if (cppHeaderOut != null) {
+    args.addAll(<String>[
+      '--objc_header_out', // TODO(gaaclarke): Switch to c++.
+      cppHeaderOut,
+    ]);
+  }
+  if (cppSourceOut != null) {
+    args.addAll(<String>[
+      '--objc_source_out', // TODO(gaaclarke): Switch to c++.
+      cppSourceOut,
+    ]);
+  }
+  if (!hasDart) {
+    args.add('--one_language');
+  }
+  final Process generate = await _streamOutput(Process.start('dart', args));
+  final int generateCode = await generate.exitCode;
+  if (generateCode != 0) {
+    return generateCode;
+  }
+  return 0;
+}
+
+Future<int> _runWindowsUnitTests() async {
+  const String windowsUnitTestsPath = './platform_tests/windows_unit_tests';
+  final int generateCode = await _runPigeon(
+    input: './pigeons/message.dart',
+    cppHeaderOut: '$windowsUnitTestsPath/windows/test/message.g.h',
+    cppSourceOut: '$windowsUnitTestsPath/windows/test/message.g.cpp',
+  );
+  if (generateCode != 0) {
+    return generateCode;
```

Comment:
Error messages at the specific failure point would help with understanding/debugging failures.

Reply:
Same thing, stdout/stderr streaming should cover that.